### DISCUSSION TITLE
Discovering and running MSBuild

### DISCUSSION THREAD

**Discussion by apodtele** [git-for-windows/git#3507] [Q&A] [answered]
GfW is not MSYS2, but I want to run MSBuild. It works if I just add the path to the executable as [described](https://jamesbrucecodes.wordpress.com/2019/03/04/running-msbuild-on-windows-in-git-bash/).

1) Can this path be discovered and added during installation?
2) Does it belong to new /etc/profile.d/msbuild.sh?

Please advise on the proper setup.

**Comment by dscho** [comment, marked as answer]
After `sdk init build-extra` (needs to be run only once), this should be available: https://github.com/git-for-windows/build-extra/tree/main/msbuild (we use this in our automation).

**Reply by apodtele** [reply]
This implies that I need Git for Windows SDK to begin with, which is advertised for your development community, not exactly for me.

I am a really minimalist developer: I use vi and git from the GfW distribution and MSBuild from Build Tools for Visual Studio. It turns out that if I just add the path, MSBuild works from Git Bash. This _almost_ matches VS Code: a good code editor, integrated git, and a build system. I am really happy after I added msbuild.sh to /etc/profile.d/
```
# Set path to MSBuild

MSBUILD="/c/Program Files (x86)/Microsoft Visual Studio/2022/BuildTools/MSBuild/Current/Bin"

[ -d "$MSBUILD" ] && PATH="$PATH:$MSBUILD"

export PATH
```
I now wonder if I can integrate the CDB debugger as well.

**Reply by dscho** [reply]
>  I added msbuild.sh to /etc/profile.d/

You will find that it works better across Git for Windows upgrades if you move this into `~/.bash_profile`.

**Reply by apodtele** [reply]
I followed your advise and also added an alias with color.
```
alias msbuild='MSBuild.exe -clp:ForceConsoleColor'
```

**Reply by dscho** [reply]
Nice! I'm sure this discussion will help others who are in the same boat as you. Thanks for starting it!
